ISS-189649:  Update the mobile SDK documentation

dhananjayj-dev opened the pull request:
## Summary
<!-- Provide a brief description of the story behind this PR, as if explaining
     to a non-technical person. Or to an LLM so it can learn from it for
     future (autonomous) code improvements. Feel free to point to a deeper
     design doc, if applicable.
-->

This PR updates the public documentation to reflect the newly introduced custom masking architecture in the DevRev SDK.
It also fixes a number of inconsistencies across platforms.

## Connected Issues
<!-- Have you cared to connect this PR to a work item in DevRev, so that we
     can understand future routing and attribution?
-->
- [ISS-189649](https://app.devrev.ai/devrev/works/ISS-189649)

## Craftsmanship, Integrity, and Devil’s Advocacy
<!-- Mark the appropriate option with an "x" -->
- [ ] Testing: Negative test cases: null or default values, crash and fault injection tests
- [ ] Testing: Boundary conditions: rolling upgrades, denial-of-service, etc.
- [ ] Testing: Fixing a few existing — flaky or permanently-broken — test cases
- [ ] Observing: Detailed error codes so machines can understand
- [ ] Observing: Adding superior metrics for future debugging
- [ ] Observing: Tracing the hairiest pathways for field serviceability
- [ ] Training: KnowledgeOps update? So AI always works.

## Story of the craft
<!-- *“Wisdom is knowing the right path to take. Integrity is taking it.”*

    Show-n-tell the story of testing, observing, or training from above.
    Take your time. Your stories will teach others the craft.
-->

dhananjayj-dev commented on fern/docs/pages/sdks/mobile/flutter/quickstart.mdx:
At other places there is an empty line.

bojan replied:
I checked everything, you are correct. Resolving.

dhananjayj-dev commented:
@bc-devrev @bojan - Just FYI, temporarily I have used `code` icon as the original flutter icon issue is raised and it will probably be resolved next week.
<img width="343" height="139" alt="Screenshot 2025-08-01 at 1 27 18 PM" src="https://github.com/user-attachments/assets/a11664bd-0d4a-4524-bea1-16a19c357478" />

bojan commented on fern/docs/pages/sdks/mobile/android/features.mdx:
```diff
@@ -630,6 +630,74 @@ For example:
+
+For advanced use cases, you can provide a custom masking provider to explicitly specify which regions of the UI should be masked during snapshots.
+
+You can implement your own masking logic by creating a class that implements the `MaskLocationProvider` interface and setting it via `DevRev.setMaskLocationProvider(MyMaskingProvider())`. This allows you to define exact areas to mask or skip snapshots entirely for privacy and security.
```
We should use signature methods not implementations.

bc-devrev reviewed (approved):
Approved but please address the use of future tense before merging. https://www.notion.so/devrev/Why-should-I-avoid-future-tense-540a9f3531594769a1f686fdf2f4f9cf?source=copy_link

dhananjayj-dev commented:
@bojan - I’ve added documentation for token exchange and identification, updates related to push notifications in the Cordova sample app and addressed the future tense occurrences. Please re-review it.

bojan reviewed (approved):
Can we just make sure we bring the changes to the readmes?

rohan-devrev commented on fern/docs/pages/sdks/mobile/android/features.mdx:
```diff
@@ -67,16 +67,67 @@ For example:
+The steps to identify a verified user are as follows:
+1. Generate an AAT for your system (preferably through your backend).
+2. Exchange your AAT for a session token for each user of your system.
+3. Pass the user identifier and the exchanged session token to the `DevRev.identifyVerifiedUser(_:sessionToken:)` method.
```
This looks like an iOS method signature. `DevRev.identifyVerifiedUser(sessionToken: String)` can be used instead.

rohan-devrev commented on fern/docs/pages/sdks/mobile/android/features.mdx:
```diff
@@ -135,7 +186,80 @@ You can perform a logout of the current user by calling the following method:
+
+### Identity model
+
+The `Identity` class is used to provide user, organization, and account information when identifying users or updating their details. This class is used primarily with the `identifyUnverifiedUser(_:)` and `updateUser(_:)` methods.
```
The signatures look off here as well.

rohan-devrev commented on fern/docs/pages/sdks/mobile/android/quickstart.mdx:
```diff
@@ -2,10 +2,11 @@
+- Android Gradle Plugin 8.2 or later.
+- Gradle 8.9 or later.
+- The minimum Android API level should be 24.
+- Recommended: An SSH key configured locally and registered with [GitHub](https://docs.github.com/en/github/authenticating-to-github/connecting-to-github-with-ssh).
```
None of our dependencies are linked with GitHub or have such prerequisites, so this may be redundant for Android.

zeeshan-devrev commented on fern/docs/pages/sdks/mobile/react-native/features.mdx:
```diff
@@ -183,21 +314,34 @@ The mechanism uses balanced start and stop methods, both of which accept a timer
 
 ### Screen tracking
 
 The DevRev SDK offers automatic screen tracking to help you understand how users navigate through your app. Although view controllers are automatically tracked, you can manually track screens using the following method:
```
```suggestion
The DevRev SDK offers automatic screen tracking to help you understand how users navigate through your app. Although view controllers or activities are automatically tracked, you can manually track screens using the following method:
```

zeeshan-devrev commented on fern/docs/pages/sdks/mobile/react-native/features.mdx:
```diff
@@ -183,21 +314,34 @@ The mechanism uses balanced start and stop methods, both of which accept a timer
 
 ### Screen tracking
 
 The DevRev SDK offers automatic screen tracking to help you understand how users navigate through your app. Although view controllers are automatically tracked, you can manually track screens using the following method:
```
I believe it would be more accurate to refer to this as `screen name tracking` rather than `screen tracking`, as the latter could potentially mislead customers.

bojan replied:
I'd leave this for now as it was named like that before, we can try to think of a better name and change the APIs as well.

zeeshan-devrev reviewed (changes requested):
Small change for screen name tracking.
